doc/source/conf.py

```diff
@@ -107,10 +108,74 @@ def import_or_mock(module):
+                                                       "Callback API")
+
+# fix "more than one target found for cross-reference 'RayParams'"
+sys.modules["xgboost_ray"].train.__doc__ = sys.modules[
+    "xgboost_ray"].train.__doc__.replace("RayParams", "xgboost_ray.RayParams")
+sys.modules["xgboost_ray"].predict.__doc__ = sys.modules[
+    "xgboost_ray"].predict.__doc__.replace("RayParams",
+                                           "xgboost_ray.RayParams")
+sys.modules["xgboost_ray"].RayXGBClassifier.fit.__doc__ = sys.modules[
+    "xgboost_ray"].RayXGBClassifier.fit.__doc__.replace(
+        "RayParams", "xgboost_ray.RayParams")
+sys.modules["xgboost_ray"].RayXGBClassifier.predict.__doc__ = sys.modules[
+    "xgboost_ray"].RayXGBClassifier.predict.__doc__.replace(
+        "RayParams", "xgboost_ray.RayParams")
+sys.modules[
+    "xgboost_ray"].RayXGBClassifier.predict_proba.__doc__ = sys.modules[
+        "xgboost_ray"].RayXGBClassifier.predict_proba.__doc__.replace(
+            "RayParams", "xgboost_ray.RayParams")
+sys.modules["xgboost_ray"].RayXGBRegressor.fit.__doc__ = sys.modules[
+    "xgboost_ray"].RayXGBRegressor.fit.__doc__.replace(
+        "RayParams", "xgboost_ray.RayParams")
+sys.modules["xgboost_ray"].RayXGBRegressor.predict.__doc__ = sys.modules[
+    "xgboost_ray"].RayXGBRegressor.predict.__doc__.replace(
+        "RayParams", "xgboost_ray.RayParams")
+
+lgbm_modules = [
+    "lightgbm.main", "lightgbm.callback", "lightgbm.compat", "lightgbm_ray"
+]
+for lgbm_module in lgbm_modules:
+    import_or_mock(lgbm_module)
+
+# fix "more than one target found for cross-reference 'RayParams'"
+sys.modules["lightgbm_ray"].train.__doc__ = sys.modules[
+    "lightgbm_ray"].train.__doc__.replace("RayParams",
+                                          "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].predict.__doc__ = sys.modules[
+    "lightgbm_ray"].predict.__doc__.replace("RayParams",
+                                            "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].RayLGBMClassifier.fit.__doc__ = sys.modules[
+    "lightgbm_ray"].RayLGBMClassifier.fit.__doc__.replace(
+        "RayParams", "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].RayLGBMClassifier.predict.__doc__ = sys.modules[
+    "lightgbm_ray"].RayLGBMClassifier.predict.__doc__.replace(
+        "RayParams", "lightgbm_ray.RayParams")
+sys.modules[
+    "lightgbm_ray"].RayLGBMClassifier.predict_proba.__doc__ = sys.modules[
+        "lightgbm_ray"].RayLGBMClassifier.predict_proba.__doc__.replace(
+            "RayParams", "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].RayLGBMRegressor.fit.__doc__ = sys.modules[
+    "lightgbm_ray"].RayLGBMRegressor.fit.__doc__.replace(
+        "RayParams", "lightgbm_ray.RayParams")
+sys.modules["lightgbm_ray"].RayLGBMRegressor.predict.__doc__ = sys.modules[
+    "lightgbm_ray"].RayLGBMRegressor.predict.__doc__.replace(
+        "RayParams", "lightgbm_ray.RayParams")
+
```

Comment:
This clutters the config a bit. Can we do something like this

```
replaces = [
     (sys.modules["lightgbm_ray"].RayLGBMRegressor.fit, "RayParams", "lightgbm_ray.RayParams"),
     # ....
]
for m, s, t in replaces:
    m.__doc__ = m.__doc__.replace(s,t)
```

and add documentation before the block why this is needed and wht it achieves?

Reply:
or maybe even leave out the last two items of the tuple

```
replaces = [
     sys.modules["lightgbm_ray"].RayLGBMRegressor.fit,
    sys.modules["lightgbm_ray"].RayLGBMRegressor.predict,
     # ....
]
for m in replaces:
    m.__doc__ = m.__doc__.replace("RayParams", "lightgbm_ray.RayParams")
```